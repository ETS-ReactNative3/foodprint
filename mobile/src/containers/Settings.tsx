import React from 'react';
import { View, Text } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';

interface Props { }

const Settings: React.FC<Props> = () => {
    return (
        <SafeAreaView>
            <Text>Settings</Text>
        </SafeAreaView>
    );
};

export default Settings;
