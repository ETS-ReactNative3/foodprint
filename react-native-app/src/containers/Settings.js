import React from 'react';
import { SafeAreaView } from 'react-native-safe-area-context';
import { AuthContext } from '../store/Auth';
import { useContext } from 'react';
import { Avatar, Button } from 'react-native-elements';
import { View } from 'react-native';

const Settings = ({ navigation }) => {

  const { signOut } = useContext(AuthContext);

  return (
    <SafeAreaView style={{ flex: 1 }}>
      <View style={{ flex: 1.5, alignItems: 'center', justifyContent: 'center' }}>
        <Avatar
          size={'xlarge'}
          rounded icon={{ name: 'home' }}
          onPress={() => console.warn('Implement "Change avatar"')}
        />
      </View>
      <View style={{ flex: 1, justify: 'flex-start', alignItems: 'flex-start', marginHorizontal: 50 }}>
        <Button
          buttonStyle={{ flexDirection: 'row', justifyContent: 'flex-start' }}
          titleStyle={{ color: 'black', fontSize: 18 }}
          title="Change username"
          type="clear"
          onPress={() => console.warn('Implement "Change username"')}
        />
        <Button
          buttonStyle={{ flexDirection: 'row', justifyContent: 'flex-start' }}
          titleStyle={{ color: 'black', fontSize: 18 }}
          title="Change email address"
          type="clear"
          onPress={() => console.warn('Implement "Change email address"')}
        />
        <Button
          buttonStyle={{ flexDirection: 'row', justifyContent: 'flex-start' }}
          titleStyle={{ color: 'black', fontSize: 18, textAlign: 'left' }}
          title="Change password"
          type="clear"
          onPress={() => console.warn('Implement "Change password"')}
        />
        <Button
          buttonStyle={{ flexDirection: 'row', justifyContent: 'flex-start' }}
          titleStyle={{ color: 'black', fontSize: 18 }}
          title="Log out"
          type="clear"
          onPress={signOut}
        />
      </View>
    </SafeAreaView>
  );
};

export default Settings;
