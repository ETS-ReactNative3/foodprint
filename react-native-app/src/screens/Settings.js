import React from 'react';
import { SafeAreaView } from 'react-native-safe-area-context';
import { useContext } from 'react';
import { View } from 'react-native';
import AuthContext from '../context/AuthContext';
import { Linking } from 'react-native';
import { useState } from 'react';
import { ListItem, Text } from 'react-native-elements';
import { StyleSheet } from 'react-native';

const Settings = ({ navigation }) => {
  const [errors, setErrors] = useState([]);
  const { signOut } = useContext(AuthContext);

  const openEmail = () => {
    Linking.openURL('mailto:ben@fresla.co?subject=Foodprint%20%20Support').catch(
      err => {
        console.warn(err);
        setErrors(['Unable to open mail app, do you have one set?']);
      },
    );
  };

  return (
    <SafeAreaView>
      <View style={styles.titleContainer}>
        <Text style={styles.title}>{'Settings'}</Text>
        <Text>let us know if you need help.</Text>
      </View>
      <View style={styles.listContainer}>
        <ListItem
          containerStyle={styles.buttonContainer}
          title="Email us"
          topDivider
          bottomDivider
          onPress={openEmail}
        />
        <ListItem
          containerStyle={styles.buttonContainer}
          title="Delete account"
          bottomDivider
          onPress={() => navigation.navigate('Delete Account')}
        />
        <ListItem
          containerStyle={styles.buttonContainer}
          title="Log out"
          bottomDivider
          onPress={signOut}
        />
      </View>
      <View>
        {errors.map((error, i) => (
          <Text key={i}>Error: {error}</Text>
        ))}
      </View>
    </SafeAreaView>
  );
};

export default Settings;

const styles = StyleSheet.create({
  modalContainer: { padding: 16 },
  titleContainer: {
    margin: 8,
  },
  title: {
    fontSize: 24,
  },
  listContainer: {
    width: '100%',
  },
  buttonContainer: {
    marginTop: 0,
    width: '100%',
  },
  button: {
    height: 48,
  },
});

