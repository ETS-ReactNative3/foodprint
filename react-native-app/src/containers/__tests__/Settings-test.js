import React from 'react';

import renderer from 'react-test-renderer';
import { AuthContext } from '../../store/Auth';
import Settings from '../Settings';
import { SafeAreaProvider } from 'react-native-safe-area-context';

jest.mock('@apollo/react-hooks');

// Check current component against its snapshot
test('ErrorMessage renders correctly and is the same as its previous snapshot', () => {
  const authentication = {
    signOut: () => { },
  };

  const tree = renderer.create(<AuthContext.Provider value={authentication}>
    <SafeAreaProvider>
      <Settings />
    </SafeAreaProvider>
  </AuthContext.Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
