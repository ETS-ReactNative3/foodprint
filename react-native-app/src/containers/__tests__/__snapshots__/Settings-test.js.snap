// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage renders correctly and is the same as its previous snapshot 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
